.github/workflows/lint-workflow-files.yml

```diff
@@ -0,0 +1,25 @@
+name: Lint GitHub Workflow YAML Files
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/*.yml'
+      - '.github/workflows/*.yaml'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Lint GitHub Workflow YAML Files with Docker
+        run: |
+          docker run --rm -v "${{ github.workspace }}:/repo" --workdir /repo rhysd/actionlint:latest -color .github/workflows/*.yml .github/workflows/*.yaml
```

Comment:
Should this use a specific version rather than `latest` to prevent the build from suddenly breaking with an unexpected change in a new release?

(ideally, the version should also be auto-upgraded with a renovate PR)

Reply:
You're right using a specific version will ensure stability. I will update that. thanks

Reply:
Changes made here https://github.com/open-telemetry/opentelemetry-collector/pull/11493/commits/f4c89fdab2d4d8988d027bf1db9bdff80574adf5 and https://github.com/open-telemetry/opentelemetry-collector/pull/11493/commits/875f2c3d276403b91caf4e87f2dee82a713d81e2